### ISSUE TITLE
Allow for subfiles in `ext`

### ISSUE THREAD

**Issue by pat-alt** [cjdoris/PackageExtensionCompat.jl#12]
You currenlty cannot split and organise the files in `ext` into various subfiles. I originally did this in this [PR](https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/pull/270):

```
ext
├── MPIExt
│   ├── MPIExt.jl
│   └── utils.jl
├── PythonCallExt
│   ├── PythonCallExt.jl
│   ├── models.jl
│   └── utils.jl
└── RCallExt
    ├── RCallExt.jl
    └── utils.jl
```

Using this setup tests failed on Julia versions `<1.9` after [warnings](https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/actions/runs/6026392570/job/16349210977#step:6:1506) such as the one below that certain extension files couldn't be located:

```julia
┌ Warning: Error requiring `MPI` from `CounterfactualExplanations`
│   exception =
│    LoadError: SystemError: opening file "/home/runner/work/CounterfactualExplanations.jl/CounterfactualExplanations.jl/ext_compat/MPIExt/utils.jl": No such file or directory
│    Stacktrace:
[...]
```

Eventually, I [moved](https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/pull/270/commits/139aae01dda74be059b4ea6fa29a5a9ed267cc4f) all code into single files

```
ext
├── MPIExt.jl
├── PythonCallExt.jl
└── RCallExt.jl
```

and all tests passed, but the previous folder structure was more organised.

_Originally posted by @pat-alt in https://github.com/cjdoris/PackageExtensionCompat.jl/issues/11#issuecomment-1700391075_

**Comment by cjdoris** [comment]
Thanks yeah. I should also just document the current limitations (of which I have a couple in my head).

**Comment by cjdoris** [comment]
I believe this is fixed by #11 (please comment if not).

**Comment by pat-alt** [comment]
Can confirm this is now working, thank you!
